addons/godot-next/references/MenuManager.gd

```diff
@@ -0,0 +1,76 @@
+class_name MenuManager
```

Comment:
I'm not sure I like the name `MenuManager` for what this class does. Usually, a "Manager" name is reserved for some sort of top-level singleton that manages a large range of objects at a high level. This is all about organizing the order of just one menu.

In fact, I would think you could drastically change the design of this class to become much more user friendly by just creating a Control node that's explicit purpose is creating a list of user-defined nodes/scenes and creating a menu-like interaction with them. Similar to PopupMenu, but without it being a popup. And in that case, rather than being a `MenuManager` (which I would think of as an entity that creates and manages multiple menus, with clean transitions, saving data, etc.), it would be more like a `MenuPanel` or something.

Perhaps with a MarginContainer inside it that allows the user to configure margins for the `MenuPanel` to decide how far inset the menu items should be relative to the panel background. And/or perhaps have a toggle for whether the panel itself should render at all (`$Panel.self_modulate.a = 0.0 if hide_background else 1.0`).

Another feature idea would be to have them be tool scripts that automatically update with a preview of what they will render as you modify their exported properties. Lots of potential here.

Now, if you want to keep it as a Reference class that just calculates how a single menu could update in response to inputs, I'd probably change the name to `MenuInputHandler` or something. That is a lot more precise in its intent and feature set, without being as likely to infringe on class names that others use in their own applications.

Anyway, these are just ideas. Let me know what you think.

Reply:
Well, this would completely changes the objective of this class, I made it to "Manage Menus" for me, this is not a control derived class, it manages each menu of the game, and the visual is up to the programmer (stated in the class description too).